### PR TITLE
LedgerCloseMetaV2

### PR DESCRIPTION
**Pull request by paulbellamy**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Implement basic `ingest` package support for LedgerCloseMetaV2. Note this doesn't do anything for horizon.

### Why

So the soroban-rpc transaction unit tests stop failing.

### Known limitations

[TODO or N/A]

### CODE REVIEW

File: xdr/ledger_close_meta.go [hunk 1]
```diff
@@ -58,6 +62,16 @@ func (l LedgerCloseMeta) TransactionEnvelopes() []TransactionEnvelope {
 			}
 		}
 		return envelopes
+	case 2:
+		var envelopes = make([]TransactionEnvelope, 0, l.CountTransactions())
+		var phases []TransactionPhase
+		phases = l.MustV2().TxSet.V1TxSet.Phases
```
Comment (tsachiherman) [inline xdr/ledger_close_meta.go:68]: The `V1TxSet` is a pointer; to ensure it's a non-nil, could you add an if clause ?

File: xdr/ledger_close_meta.go [hunk 2]
```diff
@@ -58,6 +62,16 @@ func (l LedgerCloseMeta) TransactionEnvelopes() []TransactionEnvelope {
...
+		for _, phase := range phases {
+			for _, component := range *phase.V0Components {
```
Comment (tsachiherman) [inline xdr/ledger_close_meta.go:70]: The `V0Components` are a pointer to `[]TxSetComponent`; could you add an if to ensure it's non-nil ?

File: xdr/ledger_close_meta.go [hunk 3]
```diff
@@ -70,6 +84,8 @@ func (l LedgerCloseMeta) TransactionHash(i int) Hash {
 		return l.MustV0().TxProcessing[i].Result.TransactionHash
 	case 1:
 		return l.MustV1().TxProcessing[i].Result.TransactionHash
+	case 2:
+		return l.MustV2().TxProcessing[i].Result.TransactionHash
```
Comment (tsachiherman) [inline xdr/ledger_close_meta.go:88]: what happen if i >= len(l.MustV2().TxProcessing) ?

**Review by tsachiherman — CHANGES_REQUESTED** [review]
Few small requests.

File: xdr/ledger_close_meta.go (hunk 1 above)
Comment (paulbellamy) [reply xdr/ledger_close_meta.go:68]: It's also the only case in the "union".

File: xdr/ledger_close_meta.go (hunk 3 above)
Comment (paulbellamy) [reply xdr/ledger_close_meta.go:88]: a panic

**Review by tsachiherman — APPROVED** [review]

**Pull request merged**